### ISSUE TITLE
Add progress bar/percentage displayed for importing content

### ISSUE THREAD

**Issue by contempoinc** [richtabor/MerlinWP#55]
@richtabor @capuderg @primozcigler The loading icon is great for quick imports but when you're dealing with larger ones it gives the user no idea of where they're really at, which is just plain bad for UX. I think adding a nice progress bar or percentage on each of the items "Content", "Widgets", "Sliders", etc…example:

<a href="https://cl.ly/27040c3K2I28" target="_blank"><img src="https://d3vv6lp55qjaqc.cloudfront.net/items/2d1y1Y3w0L3S0q2D1w3k/Screen%20Recording%202018-03-06%20at%2009.13%20AM.gif" style="display: block;height: auto;width: 100%;"/></a>

**Comment by contempoinc** [comment]
@primozcigler @capuderg Same with the other feature request I'm willing to invest some funds for this as well.

**Comment by contempoinc** [comment]
Any updates on this?

**Comment by capuderg** [comment]
We have it on the todo list, but I can't give you any specific dates as to when this will be added.

**Comment by capuderg** [comment]
@richtabor I'll be taking a closer look at this later today and tomorrow, do you have any UX ideas, how the progress should be displayed?

The item that takes the most time is the content import, others complete relatively quickly.

Should there be a progress bar below all import items (one progress bar for all items) and the content import would have a "weight" of 75%?

**Comment by capuderg** [comment]
I've just pushed a working progress bar, with basic styling in this branch: `add-content-import-progress-bar`

I've added the progress bar below the import items, so it will be visible even, if you don't open the "advanced" section.

- The progress bar will be displayed only, if the "Content" task is selected for import. 
- The progress is displayed and calculated only with the data from the content importer (widget, customizer options and other tasks are ignored). The content part of the import takes more then 90% of the importing time (if not even more), so I just ignored those parts.

@richtabor  I would like to get some feedback, if the progress bar is functioning OK for you. And your input into design of the progress bar.

**Comment by richtabor** [comment]
I just pushed a commit that moves the progress bar up above. What do you think?

<img width="692" alt="screen shot 2018-04-15 at 11 03 50 am" src="https://user-images.githubusercontent.com/1813435/38779889-cca0a562-409c-11e8-964a-a1c597e7c7a1.png">

**Comment by richtabor** [comment]
@contempoinc Have you been able to test out the feature yet?

**Comment by capuderg** [comment]
@richtabor would it be better to position the bar below the "import" button (at the bottom of the white "box").

So, if you are importing this on mobile, or on smaller screens, you might not see the top of the "box"?

Just a thought... I like it otherwise.

**Comment by primozcigler** [comment]
The progress bar at the top looks slick, but it lacks good UX in my opinion. It's not apparent enough. Also, I'd include the `XX %` somewhere if possible. It's very important in this step that the user is very confident that the importer is working and something's going on.

**Comment by richtabor** [comment]
Maybe we should simply remove the spinner that replaces the Import button, and instead add a radial progress bar in place of it (with XX% if possible). What do you think @capuderg @primozcigler?

**Comment by contempoinc** [comment]
I think adding a radial progress bar with XX% would make the most sense.

**Comment by redlagoon** [comment]
tl;dr: The line is awesome!

I'm too smart version:

I believe that, due to the familiarization of most users (non-blog users) with the WordPress eco-system and as well as with other sites, the green progress bar works best. YouTube uses it. Google uses it. Facebook uses it, as such, we already have a base for what is "widely accepted and understood".

Not saying copy these companies' features, but...

At first I was convoluted because, due to its position, it may stir confusion that is called "the connector dillema" where people don't know what one piece connects to, even you can clearly tell that it's doing something.

In regards to blog themes users, I've found they're the least tech-savy, but they get around, but I still strongly believe that the whole UI is so dead-simple there's no way you could go wrong.

**Comment by contempoinc** [comment]
Problem with the top bar it changes the already existing and familiar UI of replacing the "Next" button with a loader, so for consistency that's where a circular loader with an XX% in the middle makes the most sense.

**Comment by redlagoon** [comment]
Got it. I just saw some screens and assumed it is what it is for 99% of the time. I'll load it up and actually experiment, but if we're talking about a load-line, it's the same good old, right?

**Comment by capuderg** [comment]
@richtabor yes, your idea works great!

Let me know, if you need help with anything.

**Comment by contempoinc** [comment]
Any word on this one?

**Comment by richtabor** [comment]
@contempoinc @capuderg @primozcigler What do you all think of this? 

The green background of the button is the progress bar, and the % updates as well.

<img width="430" alt="screen shot 2018-06-08 at 2 38 10 pm" src="https://user-images.githubusercontent.com/1813435/41174962-9a4ad328-6b29-11e8-9054-3983fc8abfe1.png">
<img width="669" alt="screen shot 2018-06-08 at 2 34 06 pm" src="https://user-images.githubusercontent.com/1813435/41174853-40ab3fd8-6b29-11e8-958a-733d458463c8.png">

I tried working in a circular loader, but it was getting to be too tricky without adding another JS dependency (which I wanted to avoid).

**Comment by contempoinc** [comment]
@richtabor Love it!

**Comment by capuderg** [comment]
@richtabor this looks great, however the progress is only calculated for the content import, so the 100% will be displayed while the widgets and customizer are still importing. If that's not an issue, then it's all good.

Or we can cap it at 99 when the content import is done and set it to 100 at the end of the import.

**Comment by richtabor** [comment]
@capuderg Makes sense!

**Comment by richtabor** [comment]
Merged in PR #109 🎉
